cmd/cluster/validatepullsecret.go

```diff
@@ -0,0 +1,100 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var BackplaneClusterAdmin = "backplane-cluster-admin"
+
+func newCmdValidatePullSecret(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "validate-pull-secret",
+		Short:             "Checks if the pull secret email matches the owner email",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+		},
+	}
+}
+
+func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+	ocm := utils.CreateConnection()
+	defer func() {
+		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
+		}
+	}()
+
+	fmt.Println("Checking if pull secret email matches user email")
+
+	// This is the flagset for the kubeCli object provided from the root command. Set here to retroactively impersonate backplane-cluster-admin
+	flags.Impersonate = &BackplaneClusterAdmin
+	secret := &corev1.Secret{}
+	err := kubeCli.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-config", Name: "pull-secret"}, secret)
+	if err != nil {
+		return err
+	}
+
+	type auth struct {
+		Email string `json:"email"`
+	}
+
+	type auths struct {
+		Cloudopenshiftcom auth `json:"cloud.openshift.com"`
+	}
+
+	type response struct {
+		Auths auths `json:"auths"`
+	}
+
+	dockerConfigJsonBytes, found := secret.Data[".dockerconfigjson"]
```

Comment:
I still there there's likely somewhere we can pull this definition from. I'm going to take a look and see if I can find it. 

If we can't find it, then I propose we should move this out of the function and at least define these at the package level, but potentially at another level. That way, any other script or logic needed to access this same json can use that definition, rather than redefining it each time we need it

Reply:
@fahlmant the struct we want is https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/access_token_type.go#L23

However, none of the fields are exported so I wasn't able to unmarshal a response to that struct. I agree that if there's a way we should be using it, and I'm open to input.

Reply:
It looks like we might still be able to use the SDK functions to Unmarshal and access fields
https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/access_token_auth_type_json.go#L67
https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/access_token_auth_type.go#L55

Reply:
Good find here! I think https://github.com/openshift-online/ocm-sdk-go/blob/main/accountsmgmt/v1/access_token_type_json.go#L79 will get us what we need here.